theano/tensor/tests/test_opt.py

```diff
@@ -6656,6 +6656,54 @@ def test_local_useless_alloc():
     assert isinstance(topo[-1].op, T.Alloc)
 
 
+def test_local_log_sum_exp1():
+    # Tests if optimization is applied
+    x = matrix('x')
+    y = T.log(T.sum(T.exp(x), axis=1))
```

Comment:
We should test with:
- axes other than the last
- multiple axes
- sum with and without `keepdims`
- more dimensions than 2, and a dimshuffle after the sum, for instance `T.log(T.sum(T.exp(t3), axis=1).T)`

And make sure the tests pass with DebugMode.